src/tests/ftest/control/dmg_telemetry_io_latency.py

```diff
@@ -339,14 +340,132 @@ def test_io_latency_telmetry_metrics(self):
         if errors:
             self.fail("Test FAILED")
 
+    def verify_io_dtx_committed_metrics(self, metrics_data, test_metrics, transfer_size):
+        """Verify IO dtx metrics after running IOR.
+
+        Args:
+            metrics_data (dict): dictionary of io dtx "committed" metrics
+            test_metrics (dict): io dtx "committed" telemetry metrics
+            transfer_size(str): transfer_size used with ior
+
+        Returns:
+            status: (bool) True if metrics are verified for transfer size
+
+        """
+        block_size = self.params.get("block_size", "/run/*")
+        repetitions = self.params.get("repetitions", "/run/*")
+        test_ops = 3
+        status = True
+
+        metrics = {}
+        for test_metric in test_metrics:
+            metrics[test_metric] = 0
+            for host in self.hostlist_servers:
+                # test assumes one engine per host
+                for rank in self.server_managers[-1].get_host_ranks([host]):
+                    for target in range(self.server_managers[-1].get_config_value("targets")):
+                        value = metrics_data[test_metric][host][str(rank)][str(target)]["-"]
+                        metrics[test_metric] = metrics[test_metric] + value
+        dtx_value = metrics["engine_io_dtx_committed"]
+        min_value = metrics["engine_io_dtx_committed_min"]
+        max_value = metrics["engine_io_dtx_committed_max"]
+        mean_value = metrics["engine_io_dtx_committed_mean"]
+        stddev_value = metrics["engine_io_dtx_committed_stddev"]
+
+        # DTX committed value should be equal to the number of transactions.
+        if (transfer_size == "512"):
+            transfer_size = "512B"
+
+        # num operations = (blocksize / xfersize) * repetitions + ops for create/remove
+        num_operations = convert_to_number(block_size) / convert_to_number(transfer_size)
+        num_operations *= repetitions
+        # test ops = (create dir + punch obj + punch dir) * repetitions
+        test_ops *= repetitions
+        num_operations += test_ops
+
+        if (dtx_value != num_operations):
+            self.log.error("engine_io_dtx_committed NOT verified, %s != %s",
+                           dtx_value, num_operations)
+            status = False
+        if (min_value != 0):
+            self.log.error("engine_io_dtx_committed_min != 0")
+            status = False
+        if not ((max_value >= dtx_value >= min_value) and (
+                max_value > mean_value > min_value) and (
+                    stddev_value < (max_value-min_value))):
+            status = False
+
```

Comment:
The telemetry data for "io_dtx_committed" is based on VOS internal committed DTX table. From the client perspective, when the application does an update operation against a 3-ways replicated object, it will generate one DTX entry in the committed DTX table on every replica. So there are total 3 DTX entries in this example for the whole servers (with 3 replicas). For punch akey/dkey, it is similar as update case. But for punch object, the case may be some complex, because the 3-ways replicated object may have multiple redundancy groups (for 3G2 or 3GX cases), the punch object operation will generate one DTX entry on each replica in every redundancy group.

On the other hand, as time going, if exceeds the DTX aggregation time threshold (630 seconds by default), then DTX aggregation will be triggered and some DTX entries will be removed from the committed DTX table. Under such case, the io_dtx_committed max count may does not match the the expected "num_operations".

So just reminder to correctly calculate the expected "num_operations".

Reply:
Another thing to be noticed is that by default the DTX entries for update and punch will be batched committed on leader asynchronously. The batched internal is 10 seconds at most. Means that if you immediately check the io_dtx_committed value after update/punch, you may get smaller value than the expected number. So if you want to get the final result, you may have wait for sometime, about 10 seconds.

Reply:
@Nasf-Fan Sorry a little confused. There is only 1 target in this example with the object class SX. So are the num_operations not being calculated correctly?

Also I can add a sleep to the code, however I am getting the expected results with the tests currently with no sleep (don't want to delay the test if not needed).

Reply:
OK, just noticed the test is done via single target. Then SX is equal to S1, all transactions will be committed synchronously after local modification without "committable" phase. So "sleep" is unnecessary under such case.